src/argilla/client/sdk/datasets/api.py

```diff
@@ -14,19 +14,20 @@
 #  limitations under the License.
 
 from functools import lru_cache
-from typing import Union
+from typing import List, Optional, Union
 
 import httpx
 
+from argilla._constants import WORKSPACE_HEADER_NAME
 from argilla.client.sdk.client import AuthenticatedClient
 from argilla.client.sdk.commons.errors_handler import handle_response_error
 from argilla.client.sdk.commons.models import ErrorMessage, HTTPValidationError, Response
 from argilla.client.sdk.datasets.models import CopyDatasetRequest, Dataset
 
 
 @lru_cache(maxsize=None)
-def get_dataset(client: AuthenticatedClient, name: str) -> Response[Dataset]:
-    url = "{}/api/datasets/{name}".format(client.base_url, name=name)
+def get_dataset(client: AuthenticatedClient, name: str) -> Response[Union[Dataset, ErrorMessage, HTTPValidationError]]:
```

Comment:
Errors occurred here will be raised instead of returned as part of a response.
```suggestion
def get_dataset(client: AuthenticatedClient, name: str) -> Response[Dataset]:
```